pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -314,24 +314,25 @@ func (f *framework) RunFilterPlugins(pc *PluginContext,
 }
 
 // RunScorePlugins runs the set of configured scoring plugins. It returns a map that
-// stores for each scoring plugin name the corresponding NodeScoreList(s).
+// stores for each scoring plugin name the corresponding NodeScoreMap(s).
 // It also returns *Status, which is set to non-success if any of the plugins returns
 // a non-success status.
 func (f *framework) RunScorePlugins(pc *PluginContext, pod *v1.Pod, nodes []*v1.Node) (PluginToNodeScoreMap, *Status) {
 	pluginToNodeScoreMap := make(PluginToNodeScoreMap, len(f.scorePlugins))
 	for _, pl := range f.scorePlugins {
-		pluginToNodeScoreMap[pl.Name()] = make(NodeScoreList, len(nodes))
+		pluginToNodeScoreMap[pl.Name()] = make(NodeScoreMap, len(nodes))
 	}
 	ctx, cancel := context.WithCancel(context.Background())
 	errCh := schedutil.NewErrorChannel()
 	workqueue.ParallelizeUntil(ctx, 16, len(nodes), func(index int) {
 		for _, pl := range f.scorePlugins {
+			nodeName := nodes[index].Name
 			score, status := pl.Score(pc, pod, nodes[index].Name)
 			if !status.IsSuccess() {
 				errCh.SendErrorWithCancel(fmt.Errorf(status.Message()), cancel)
 				return
 			}
-			pluginToNodeScoreMap[pl.Name()][index] = score
+			pluginToNodeScoreMap[pl.Name()][nodeName] = score
```

Comment:
I suspect you need to synchronize access when writing to the map (a pre-allocated slice doesn't) since this practically adds a new key-value to it.